src/exec-child.js

```diff
@@ -0,0 +1,42 @@
+var childProcess = require('child_process');
+var fs = require('fs');
+
+// Note: this will break if `paramFilePath` contains special characters ( '\n',
+// '\t', etc.). Such characters are possible if $TMP gets modified. We already
+// rely on tempdir() to work for other things, so this is an acceptable risk.
+var paramFilePath = process.argv[2];
+
+var serializedParams = fs.readFileSync(paramFilePath, 'utf8');
+var params = JSON.parse(serializedParams);
+
+var cmd = params.command;
+var execOptions = params.execOptions;
+var pipe = params.pipe;
+var stdoutFile = params.stdoutFile;
+var stderrFile = params.stderrFile;
+var codeFile = params.codeFile;
+
+var c = childProcess.exec(cmd, execOptions, function (err) {
+  if (!err) {
+    fs.writeFileSync(codeFile, '0');
+  } else if (err.code === undefined) {
+    fs.writeFileSync(codeFile, '1');
+  } else {
+    fs.writeFileSync(codeFile, err.code.toString());
+  }
+});
+
+var stdoutStream = fs.createWriteStream(stdoutFile);
+var stderrStream = fs.createWriteStream(stderrFile);
+
+c.stdout.pipe(stdoutStream, { end: false });
+c.stderr.pipe(stderrStream, { end: false });
+c.stdout.pipe(process.stdout);
+c.stderr.pipe(process.stderr);
+
+if (pipe) {
+  c.stdin.end(pipe);
+}
+
+c.stdout.on('end', stdoutStream.end);
+c.stderr.on('end', stderrStream.end);
```

Comment:
If this file is `require`'d, it will actually do something. To prevent this, wrap the contents like this:
```
if (require.main === module) {
    // everything goes here
}
```
This way, the code is only executed when the file is the entrypoint, like when `node exec-file.js` is run.

Reply:
Seems reasonable. Do you think it's worth adding this to similar files (e.g. `bin/shjs`)?

Reply:
Yes, I think this would be a good idea for other executable-only files.

Reply:
Filed #789. I think it would be better to fail loudly in this case (with an exception). Take a look at my alternate suggestion, let me know if there's some issue with this.

Reply:
Failing loudly sounds like a good idea.